### ISSUE TITLE
Buggy profile detection?

### ISSUE THREAD

**Issue by 1earch** [phillipberndt/autorandr#119]
Hi!

I've just discovered this wonderful script and began to use it for my laptop.
However, I think the `--detected` option has a bug...

I have 2 profiles:

- scr1 => depends on eDP1 (integrated display)
- scr2 => depends on eDP1 and DP1 (external)

If I well understood the goal of the `--detected` option, I would normally see the following output in the following cases:

1. No external display plugged
```bash
$ autorandr --detected
scr1
```
2. External display plugged
```bash
$ autorandr --detected
scr1
scr2
```

But in the second case, only the `scr2` profile is displayed. The `scr1` is also compatible with this configuration in my mind, because the display is plugged but not used.

I think this behaviour is caused by a mistake on line 558 of the script, which is:
```python
if not matches or any((name not in config.keys() for name in current_config.keys() if current_config[name].edid)):
  continue
```
but would be, in my mind:
```python
if not matches or any((name not in current_config.keys() for name in config.keys() if current_config[name].edid)):
  continue
```

Is this behaviour a bug?
If not, could you explain its goal?

Thanks in advance!

**Comment by phillipberndt** [comment]
Nope, that's intentional. There's no concept of "compatible" configurations in autorandr right now, a configuration always covers all outputs (e.g. "nothing is connected to DP1" is part of your scr1 profile).

This already was the case in the original autorandr, so I cannot tell you exactly what was the design rationale behind this. It does however make sense to me:

* The default use case is to have a configuration for your detached and docked notebook. The idea of the autodetection feature is to use it from a script. If you dock your notebook, it's natural to assume that you want to load the docked configuration.
* The notion of a "compatible" configuration isn't helpful most times. E.g. let's say there's a third configuration for said notebook, where you have your favourite beamer attached, and it only supports a pre-HD-times resolution, say 1024x768. That's totally compatible with when you dock your notebook. But you'd never ever want to load that configuration there.
* If you really want autorandr to show your scr1 profile while docked, you can just load it manually (`autorandr -l scr1`) and then save to a different name (`autorandr -s scr1-plugged`). That one would then show up in addition as a detected configuration. So there's a workaround for achieving what you want. On the other hand, if `--detected` would work with the concept of "compatible" configurations, users would have to manually write `block` scripts to achieve the current behaviour.
